File: src/lessons/welcome/bat/bool1/LoneTeenEntity.java
package lessons.welcome.bat.bool1;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class LoneTeenEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(loneTeen((Integer)param[0], (Integer)param[1])));
    }
  }

  /* BEGIN TEMPLATE */
  boolean loneTeen(int a, int b)
  {
    /* BEGIN SOLUTION */
    boolean teenA = a > 12 && a < 20;
    boolean teenB = b > 12 && b < 20;
    return (teenA && !teenB) || (teenB && !teenA);
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/welcome/bat/bool1/LoneTeenEntity.scala
package lessons.welcome.bat.bool1

import plm.universe.bat.BatEntity
import plm.universe.bat.BatTest

class LoneTeenEntity extends BatEntity {

    override def run() {
      import plm.core.ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
      	setTestResult(i, serialize(loneTeen(param(0).asInstanceOf[Int],param(1).asInstanceOf[Int]) ))
	  }
	}

	/* BEGIN TEMPLATE */
	def loneTeen(a:Int, b:Int):Boolean = {
	/* BEGIN SOLUTION */
	val teenA = a>12 && a<20
	val teenB = b>12 && b<20
	return  (teenA && !teenB) || (teenB && !teenA)
	/* END SOLUTION */
	}
	/* END TEMPLATE */
}
